diff -Naur linux-1423ac8bfbfb2a9d092b604c676e7a58a5fa3367/drivers/net/wireless/broadcom/brcm80211/brcmfmac/bcdc.c linux-1423ac8bfbfb2a9d092b604c676e7a58a5fa3367.brcmfmac-tos/drivers/net/wireless/broadcom/brcm80211/brcmfmac/bcdc.c
--- linux-1423ac8bfbfb2a9d092b604c676e7a58a5fa3367/drivers/net/wireless/broadcom/brcm80211/brcmfmac/bcdc.c	2017-05-15 17:23:14.000000000 +0200
+++ linux-1423ac8bfbfb2a9d092b604c676e7a58a5fa3367.brcmfmac-tos/drivers/net/wireless/broadcom/brcm80211/brcmfmac/bcdc.c	2017-07-22 18:27:58.027002060 +0200
@@ -266,7 +266,8 @@
 	if (pktbuf->ip_summed == CHECKSUM_PARTIAL)
 		h->flags |= BCDC_FLAG_SUM_NEEDED;
 
-	h->priority = (pktbuf->priority & BCDC_PRIORITY_MASK);
+//	h->priority = (pktbuf->priority & BCDC_PRIORITY_MASK);
+	h->priority = 0;
 	h->flags2 = 0;
 	h->data_offset = offset;
 	BCDC_SET_IF_IDX(h, ifidx);
